### PR TITLE
update technical docs and README

### PR DESCRIPTION
**Pull request by JaydenLiang**

### CODE REVIEW

**Comment by JaydenLiang** [comment]
Hi @ilee-fortinet , I invited you to review my PR here.

File: docs/autoscale_setting_items.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+| egress-traffic-route-table | The comma-separated list of route tables associated with any subnets, which should be configured to contain a route 0.0.0.0/0 to the primary FortiGate to handle egress traffic. | false | false |
+| enable-external-elb | Toggle ON / OFF the external elastic load balancing for device in the external-facing Autoscale scaling group(s). | false | false |
+| enable-fortianalyzer-integration | Enable FortiAnalyzer integration with the Autoscale FortiGate cluster. | false | false |
+| enable-hybrid-licensing | Toggle ON / OFF the hybrid licensing feature. | false | false |
+| enable-internal-elb | Toggle ON / OFF the internal elastic load balancing feature to allow traffic flow out the device in the Autoscale scaling groups(s) into an internal load balancer. | false | false |
+| enable-second-nic | Toggle ON / OFF the secondary eni creation on each device in the Autoscale scaling group(s). | false | false |
+| enable-vm-info-cache | Toggle ON / OFF the vm info cache feature. It caches the vm info in db to reduce API calls to query a vm from the platform. | false | false |
```
Comment (ilee-fortinet) [inline docs/autoscale_setting_items.md:49]:
```suggestion
| enable-vm-info-cache | Toggle ON / OFF the VM info cache feature. It caches the VM info in the DB to reduce API calls to query a VM from the platform. | false | false |
```

File: docs/autoscale_setting_items.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+| faz-handler-name | The FortiGate Autoscale - FortiAnalyzer handler function name. | false | false |
+| faz-ip | The FortiGate Autoscale - FortiAnalyzer ip address. | false | false |
+| fortigate-admin-port | The port number for administrative login to a FortiGate. | false | true |
+| fortigate-autoscale-setting-saved | The flag whether FortiGate Autoscale settings are saved in db or not. | false | false |
+| fortigate-autoscale-subnet-id-list | A comma-separated list of FortiGate Autoscale subnet IDs. | false | false |
+| fortigate-autoscale-subnet-pairs | A list of paired subnets for north-south traffic routing purposes. Format: [{subnetId: [pairId1, pairId2, ...]}, ...] | true | false |
+| fortigate-autoscale-virtual-network-cidr | CIDR of the Virtual Network that contains FortiGate Autoscale. | false | false |
+| fortigate-autoscale-virtual-network-id | ID of the Virtual Network that contains FortiGate Autoscale. | false | false |
```
Comment (ilee-fortinet) [inline docs/autoscale_setting_items.md:57]:
```suggestion
| fortigate-autoscale-virtual-network-id | ID of the virtual network that contains FortiGate Autoscale. | false | false |
```

File: docs/autoscale_setting_items.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+| fortigate-psk-secret | The PSK for FortiGate Autoscale synchronization. | false | true |
+| fortigate-sync-interface | The interface the FortiGate uses for configuration synchronization. | false | true |
+| fortigate-traffic-port | The port number for the load balancer to route traffic through FortiGates to the protected services behind the load balancer. | false | true |
+| fortigate-traffic-protocol | The protocol for the traffic to be routed by the load balancer through FortiGates to the protected services behind the load balancer. | false | true |
+| heartbeat-delay-allowance | The maximum amount of time (in seconds) allowed for network latency of the Autoscale device heartbeat arriving at the Autoscale handler. | false | true |
+| heartbeat-interval | The length of time (in seconds) that an Autoscale device waits between sending heartbeat requests to the Autoscale handler. | false | true |
+| heartbeat-loss-count | Number of consecutively lost heartbeats. When the Heartbeat Loss Count has been reached, the device is deemed unhealthy and fail-over activities will commence. | false | true |
```
Comment (ilee-fortinet) [inline docs/autoscale_setting_items.md:66]:
```suggestion
| heartbeat-loss-count | Number of consecutively lost heartbeats. When the heartbeat loss count is reached, the device is deemed unhealthy and failover activities commence. | false | true |
```

File: docs/autoscale_setting_items.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+| payg-scaling-group-name | The name of the PAYG auto scaling group. | false | false |
+| primary-election-timeout | The maximum time (in seconds) to wait for a primary election to complete. | false | true |
+| primary-scaling-group-name | The name of the primary auto scaling group. | false | false |
+| resource-tag-prefix | Resource tag prefix. Used on any resource that supports tagging or labeling. Such resource will be given a tag or label starting with this prefix. Also used as the name of the logical group for Autoscale resources in those cloud platforms which support such logical grouping. | false | true |
+| scaling-group-desired-capacity | PAYG Scaling group desired capacity. | false | true |
+| scaling-group-max-size | PAYG Scaling group max size. | false | false |
+| scaling-group-min-size | PAYG Scaling group min size. | false | true |
+| sync-recovery-count | The number (positive integer) of on-time heartbeat for a vm needs to send to recover from the unhealthy state. Unhealthy vm will be excluded from being candidate of primary elections. | false | true |
```
Comment (ilee-fortinet) [inline docs/autoscale_setting_items.md:75]:
```suggestion
| sync-recovery-count | The number of on-time heartbeats (as a positive integer) that a VM must send to recover from the unhealthy state. Unhealthy VMs are excluded as primary election candidates.| false | true |
```

File: docs/autoscale_setting_items.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+| terminate-unhealthy-vm | Toggle for unhealthy vm handling behaviours. Set to true to terminate unhealthy vm or set to false to keep the unhealthy vm. | false | true |
```
Comment (ilee-fortinet) [inline docs/autoscale_setting_items.md:76]:
```suggestion
| terminate-unhealthy-vm | Toggle for unhealthy VM-handling behaviors. Set to true to terminate unhealthy VMs or to false to keep the unhealthy VMs running. | false | true |
```

File: docs/autoscale_setting_items.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+| vm-info-cache-time | The vm info cache time in seconds. | false | true |
```
Comment (ilee-fortinet) [inline docs/autoscale_setting_items.md:77]:
```suggestion
| vm-info-cache-time | The VM info cache time in seconds. | false | true |
```

File: docs/autoscale_setting_items.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+| vpn-bgp-asn | The BGP Autonomous System Number used with the VPN connections. | false | true |
```
Comment (ilee-fortinet) [inline docs/autoscale_setting_items.md:78]:
```suggestion
| vpn-bgp-asn | The BGP Autonomous System Number used with VPN connections. | false | true |
```

**Review by ilee-fortinet — COMMENTED** [review]
Please also update the lower right box in the diagram to say "handler" instead of "hander".

**Comment by JaydenLiang** [comment]
> Please also update the lower right box in the diagram to say "handler" instead of "hander".

The typo in the diagram image has been fixed.

**Review by ilee-fortinet — COMMENTED** [review]
Updated wording

**Pull request merged**